internal/log/log.go

```diff
@@ -128,6 +132,37 @@ func (logger *Logger) log(severityLevel severity, args ...interface{}) {
 	logger.logln(severityLevel, args...)
 }
 
+// TODO: This duplication is quite hideous, and should be cleaned up by
+//   delegating everything to stdlib logger in a more straightforward way.
```

Comment:
Sidenote but we could also just pull in a structured logger too to fix this

Reply:
Which part?  The fundamental problem as I see it now (I mentioned this after we merged the log stuff) is that we tested it wrong.  We're testing it as if it's a black box / greenfield project.  Whereas in fact  _our_ logger is just a thin wrapper over another (already reliable) logger, to which we delegate all important work.  Our tests should reflect this by simply asserting that a mock was called at the appropriate times, and if we write them in this way they'll be much simpler.

The only piece using a different underlying logger might help with is that it might put prefix is the place we want it by default, so we wouldn't have to rewrite the prefixing logic as we've done.  

That said, I'm not against if we have a good reason.  My position was just that we shouldn't bring in deps because they have features _we might want_ in the future.  Bringing in a dep to solve a specific, current need is fine.

Btw, we could also just take the stdlib's logger default way of printing prefixes.  I personally would be fine with that.

Anyway would like to hear more of your reasoning...